iib/workers/tasks/build.py

```diff
@@ -1049,35 +1051,46 @@ def handle_rm_request(
 
         if image_is_fbc:
             log.info("Processing File-Based Catalog image")
-            fbc_dir, _ = opm_registry_rm_fbc(
-                base_dir=temp_dir,
-                from_index=from_index_resolved,
-                operators=operators,
-                binary_image=prebuild_info['binary_image'],
-                overwrite_from_index_token=overwrite_from_index_token,
-                generate_cache=False,
-            )
-
-            # rename `catalog` directory because we need to use this name for
-            # final destination of catalog (defined in Dockerfile)
-            catalog_from_db = os.path.join(temp_dir, 'from_db')
-            os.rename(fbc_dir, catalog_from_db)
 
+            # remove operator from /configs
             os.makedirs(os.path.join(temp_dir, 'from_index'), exist_ok=True)
-            # get catalog with opted-in operators
+            # get catalog with opted-in and non-opted-in operators
+            # store it in /<temp_dir>/from_index/configs
             with set_registry_token(overwrite_from_index_token, from_index_resolved, append=True):
                 catalog_from_index = get_catalog_dir(
                     from_index=from_index_resolved, base_dir=os.path.join(temp_dir, 'from_index')
                 )
-            # remove operators from from_index file-based catalog
+            # remove operators from /<temp_dir>/from_index/configs if exists
             for operator in operators:
                 operator_path = os.path.join(catalog_from_index, operator)
                 if os.path.exists(operator_path):
                     log.debug('Removing operator from from_index FBC %s', operator_path)
                     shutil.rmtree(operator_path)
-            # overwrite data in `catalog_from_index` by data from `catalog_from_db`
-            # this adds changes on not opted in operators to final
-            merge_catalogs_dirs(catalog_from_db, catalog_from_index)
+
+            # if operator is not opted in, remove from db
+            operators_in_db, index_db_path = verify_operators_exists(
+                from_index=from_index,
+                base_dir=temp_dir,
+                operator_packages=operators,
+                overwrite_from_index_token=overwrite_from_index_token,
+            )
+
+            if operators_in_db:
+                fbc_dir, _ = opm_registry_rm_fbc(
```

Comment:
I have assumed here migrate is only needed if we are making any changes in the index.db, but I am not fully sure about it. 
If there are any cases where we need to have migrate, would be happy to update this